src/inference/tests/unit/cpu_map_parser.cpp

```diff
@@ -572,18 +574,112 @@ LinuxCpuMapTestCase _1sockets_6cores_hyperthreading = {
+        {16, 1, 16, MAIN_CORE_PROC, 16, -1},       {17, 1, 17, MAIN_CORE_PROC, 17, -1},
+        {18, 1, 18, MAIN_CORE_PROC, 18, -1},       {19, 1, 19, MAIN_CORE_PROC, 19, -1},
+        {20, 0, 0, HYPER_THREADING_PROC, 0, -1},   {21, 0, 1, HYPER_THREADING_PROC, 1, -1},
+        {22, 0, 2, HYPER_THREADING_PROC, 2, -1},   {23, 0, 3, HYPER_THREADING_PROC, 3, -1},
+        {24, 0, 4, HYPER_THREADING_PROC, 4, -1},   {25, 0, 5, HYPER_THREADING_PROC, 5, -1},
+        {26, 0, 6, HYPER_THREADING_PROC, 6, -1},   {27, 0, 7, HYPER_THREADING_PROC, 7, -1},
+        {28, 0, 8, HYPER_THREADING_PROC, 8, -1},   {29, 0, 9, HYPER_THREADING_PROC, 9, -1},
+        {30, 1, 10, HYPER_THREADING_PROC, 10, -1}, {31, 1, 11, HYPER_THREADING_PROC, 11, -1},
+        {32, 1, 12, HYPER_THREADING_PROC, 12, -1}, {33, 1, 13, HYPER_THREADING_PROC, 13, -1},
+        {34, 1, 14, HYPER_THREADING_PROC, 14, -1}, {35, 1, 15, HYPER_THREADING_PROC, 15, -1},
+        {36, 1, 16, HYPER_THREADING_PROC, 16, -1}, {37, 1, 17, HYPER_THREADING_PROC, 17, -1},
+        {38, 1, 18, HYPER_THREADING_PROC, 18, -1}, {39, 1, 19, HYPER_THREADING_PROC, 19, -1},
+    },
+    {
+        {"0,20", "0", "3000000"},  {"1,21", "0", "3000000"},  {"2,22", "0", "3000000"},  {"3,23", "0", "3000000"},
+        {"4,24", "0", "3000000"},  {"5,25", "0", "3000000"},  {"6,26", "0", "3000000"},  {"7,27", "0", "3000000"},
+        {"8,28", "0", "3000000"},  {"9,29", "0", "3000000"},  {"10,30", "1", "3000000"}, {"11,31", "1", "3000000"},
+        {"12,32", "1", "3000000"}, {"13,33", "1", "3000000"}, {"14,34", "1", "3000000"}, {"15,35", "1", "3000000"},
+        {"16,36", "1", "3000000"}, {"17,37", "1", "3000000"}, {"18,38", "1", "3000000"}, {"19,39", "1", "3000000"},
+        {"0,20", "0", "3000000"},  {"1,21", "0", "3000000"},  {"2,22", "0", "3000000"},  {"3,23", "0", "3000000"},
+        {"4,24", "0", "3000000"},  {"5,25", "0", "3000000"},  {"6,26", "0", "3000000"},  {"7,27", "0", "3000000"},
+        {"8,28", "0", "3000000"},  {"9,29", "0", "3000000"},  {"10,30", "1", "3000000"}, {"11,31", "1", "3000000"},
+        {"12,32", "1", "3000000"}, {"13,33", "1", "3000000"}, {"14,34", "1", "3000000"}, {"15,35", "1", "3000000"},
+        {"16,36", "1", "3000000"}, {"17,37", "1", "3000000"}, {"18,38", "1", "3000000"}, {"19,39", "1", "3000000"},
+    },
+};
+
+LinuxCpuMapTestCase freq_1sockets_4cores = {
+    4,
+    1,
+    4,
+    {{4, 4, 0, 0}},
+    {
+        {0, 0, 0, MAIN_CORE_PROC, 0, -1},
+        {1, 0, 1, MAIN_CORE_PROC, 1, -1},
+        {2, 0, 2, MAIN_CORE_PROC, 2, -1},
+        {3, 0, 3, MAIN_CORE_PROC, 3, -1},
+    },
+    {
+        {"0", "0", "1800000"},
+        {"1", "0", "1800000"},
+        {"2", "0", "1800000"},
+        {"3", "0", "1800000"},
+    },
+};
+
+TEST_P(LinuxCpuMapFreqParserTests, LinuxCpuMapFreq) {}
+
+INSTANTIATE_TEST_SUITE_P(CPUMap,
+                         LinuxCpuMapFreqParserTests,
+                         testing::Values(freq_2sockets_20cores_hyperthreading, freq_1sockets_4cores));
```

Comment:
I think it is lack of "Pcore + Ecore" platform test case.

Reply:
updated